// @capgo/capacitor-downloader/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

const CapacitorDownloader = core.registerPlugin('CapacitorDownloader', {
    web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.CapacitorDownloaderWeb()),
});

class CapacitorDownloaderWeb extends core.WebPlugin {
    async download(options) {
        console.log('DOWNLOAD', options);
        throw new Error('Method not implemented.');
    }
    async pause(options) {
        console.log('PAUSE', options.id);
        throw new Error('Method not implemented.');
    }
    async resume(options) {
        console.log('RESUME', options.id);
        throw new Error('Method not implemented.');
    }
    async stop(options) {
        console.log('STOP', options.id);
        throw new Error('Method not implemented.');
    }
    async checkStatus(options) {
        console.log('CHECK STATUS', options.id);
        throw new Error('Method not implemented.');
    }
    async getFileInfo(options) {
        console.log('GET FILE INFO', options.path);
        throw new Error('Method not implemented.');
    }
    async getPluginVersion() {
        return { version: 'web' };
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    CapacitorDownloaderWeb: CapacitorDownloaderWeb
});

exports.CapacitorDownloader = CapacitorDownloader;
